docs/spec.md

```diff
@@ -3114,9 +3131,9 @@ constant(result_index[iota_dimension], element_type(output))`.
 
 #### Outputs
 
-| Name     | Type                                              | Constraints |
-|----------|---------------------------------------------------|-------------|
-| `output` | tensor of integer, floating-point or complex type | (C1)        |
+| Name     | Type                                                                             | Constraints |
+|----------|----------------------------------------------------------------------------------|-------------|
+| `output` | tensor of integer, floating-point or complex type or per-tensor quantized tensor | (C1)        |
```

Comment:
What does `constant(result_index[iota_dimension], element_type(output))` mean in the cast of quantized tensors? Does result_index[iota_dimension] stand for a value of storage type here or for a value of expressed type? Same for other uses of `constant(..., type)` in the spec where type can be a quantized tensor type.

Reply:
When the `type(output)` is a quantized type,   `element_type(output)` provides the `QuantizedTensorElementType`.  The intention with `constant(result_index[iota_dimension], element_type(output))`  is to create the constant with  `QuantizedTensorElementType`.

Reply:
Added an [issue](https://github.com/openxla/stablehlo/issues/1687) to  address the usage of such  `constant` meta function on `QuantizedTensorElementType`.

Reply:
"The intention [...] is to create the constant with QuantizedTensorElementType". Sure, I understood that. What I didn't understand is whether the first argument of the `constant` function takes values of storage type or expressed type.

Reply:
The first argument of `constant(result_index[iota_dimension], element_type(output))` takes value of storage type.  IMO, there was a confusion here on whether the second argument of `constant`  function  is the type of the first argument or the casted type of first argument. I was assuming the second one and was expecting the type of `constant` to match the element type of `QuantizedTensorType`. 

Upon further thinking, it seems  the second argument of `constant` is just the type of the first argument and it makes sense to use  `output[result_index] = constant(result_index[iota_dimension], storage_type(output))`. Also updated other places (`collective_permute`, `dot_general`, and `convolution`).

Reply:
"bitcast_convert(constant(result_index[iota_dimension], storage_type(output)), element_type(output))". Let's move forward with whatever's currently written in the PR and discuss this separately in #1687.